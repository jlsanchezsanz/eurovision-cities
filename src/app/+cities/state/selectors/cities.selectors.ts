import { createFeatureSelector, createSelector } from '@ngrx/store';
import { CitiesState } from '../reducers';

export const getCitiesState = createFeatureSelector<CitiesState>('cities');

export const getCities = createSelector(getCitiesState, ({ cities }) => cities);
