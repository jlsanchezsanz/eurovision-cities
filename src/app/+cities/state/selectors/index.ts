export * from './cities.selectors';
export * from './pagination.selectors';
