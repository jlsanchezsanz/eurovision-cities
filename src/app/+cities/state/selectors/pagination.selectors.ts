import { createSelector, createFeatureSelector } from '@ngrx/store';

import { paginationFeatureKey, PaginationState } from '../reducers';

export const getPaginationState = createFeatureSelector<PaginationState>(
  paginationFeatureKey
);

export const getPageNumber = createSelector(
  getPaginationState,
  (pagination) => pagination && pagination.pageNumber
);

export const getPageSize = createSelector(
  getPaginationState,
  (pagination) => pagination && pagination.size
);

export const getTotalElements = createSelector(
  getPaginationState,
  (pagination) => pagination && pagination.totalElements
);
