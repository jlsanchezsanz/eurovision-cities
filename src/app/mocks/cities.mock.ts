import { CitiesResponse } from '../models';

export const citiesResponseMock: CitiesResponse = {
  content: [
    { id: 317, name: 'Ahvaz' },
    { id: 87, name: 'Al-Jizah' },
    { id: 147, name: 'Al-Khartum Bahri' },
    { id: 227, name: 'Al-Madinah' },
    { id: 103, name: 'Aleppo' },
    { id: 47, name: 'Alexandria' },
    { id: 176, name: 'Algiers' },
    { id: 307, name: 'Allahabad' },
    { id: 302, name: 'Almaty' },
    { id: 260, name: 'Amman' },
  ],
  pageable: {
    sort: { sorted: false, unsorted: true, empty: true },
    pageSize: 10,
    pageNumber: 1,
    offset: 10,
    unpaged: false,
    paged: true,
  },
  totalPages: 34,
  totalElements: 332,
  last: false,
  first: false,
  sort: { sorted: false, unsorted: true, empty: true },
  numberOfElements: 10,
  size: 10,
  number: 1,
  empty: false,
};
