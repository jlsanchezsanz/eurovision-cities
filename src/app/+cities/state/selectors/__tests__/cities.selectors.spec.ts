import { citiesResponseMock } from '../../../../mocks';
import { getCities } from '../cities.selectors';

describe('Cities Selectors', () => {
  describe('getCities', () => {
    it('should select cities', () => {
      expect(
        getCities.projector({ cities: { 1: citiesResponseMock.content } }, 1)
      ).toMatchSnapshot();
    });
  });
});
