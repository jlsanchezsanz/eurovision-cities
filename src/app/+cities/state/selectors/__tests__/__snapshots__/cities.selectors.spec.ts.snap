// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cities Selectors getCities should select cities 1`] = `
Array [
  Object {
    "id": 317,
    "name": "Ahvaz",
  },
  Object {
    "id": 87,
    "name": "Al-Jizah",
  },
  Object {
    "id": 147,
    "name": "Al-Khartum Bahri",
  },
  Object {
    "id": 227,
    "name": "Al-Madinah",
  },
  Object {
    "id": 103,
    "name": "Aleppo",
  },
  Object {
    "id": 47,
    "name": "Alexandria",
  },
  Object {
    "id": 176,
    "name": "Algiers",
  },
  Object {
    "id": 307,
    "name": "Allahabad",
  },
  Object {
    "id": 302,
    "name": "Almaty",
  },
  Object {
    "id": 260,
    "name": "Amman",
  },
]
`;
